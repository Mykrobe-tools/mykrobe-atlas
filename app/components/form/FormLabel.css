.wrap {
  composes: wrap from 'styles/form.css';
}

.label {
  display: inline-block;
  padding: 6px 0;
}
