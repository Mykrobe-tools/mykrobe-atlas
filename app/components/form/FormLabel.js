/* @flow */

import * as React from 'react';
import PropTypes from 'prop-types';

import styles from './FormLabel.css';

class FormLabel extends React.Component {
  render() {
    const { children, htmlFor, label } = this.props;
    return (
      <label className={styles.label} htmlFor={htmlFor} key={htmlFor}>
        {children}
        {label}
      </label>
    );
  }
}

FormLabel.propTypes = {
  children: PropTypes.node,
  htmlFor: PropTypes.string,
  label: PropTypes.string.isRequired,
};

export default FormLabel;
