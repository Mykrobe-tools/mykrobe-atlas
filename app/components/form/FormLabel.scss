.wrap {
  composes: wrap from 'styles/form.scss';
}

.label {
  display: inline-block;
  padding: 6px 0;
}
