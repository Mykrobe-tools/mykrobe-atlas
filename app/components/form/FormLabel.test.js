import * as React from 'react';
import renderer from 'react-test-renderer';
import FormLabel from './FormLabel';

describe('FormLabel component snapshot', () => {
  const htmlFor = 'loremIpsum';
  const label = 'Lorem Ipsum';
  const children = '<a href="#">Dolor Sit Amet</a>';

  it('renders correctly', () => {
    const component = renderer.create(
      <FormLabel htmlFor={htmlFor} label={label}>
        {children}
      </FormLabel>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
