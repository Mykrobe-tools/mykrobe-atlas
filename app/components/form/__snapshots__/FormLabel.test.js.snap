// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLabel component snapshot renders correctly 1`] = `
<label
  className="label"
  htmlFor="loremIpsum"
>
  &lt;a href="#"&gt;Dolor Sit Amet&lt;/a&gt;
  Lorem Ipsum
</label>
`;
